### ISSUE TITLE
Allow AND logic for searching assets on same filter type

### ISSUE THREAD

**Issue by garreeoke** [kubecost/cost-analyzer-helm-chart#893]
User would like the ability to apply AND logic to items within the same filter type.

**Describe the solution you'd like**
Be able to find assets that only have two specific labels. For example, if I filter for namespace:kubecost, namespace:spinnaker I should only see results that have both.

**How would users interact with this feature?**
In the UI, would be able to add filterTypes of the same kind in a query.

**Comment by dwbrown2** [comment]
I think this a good request that would be consistent with the Allocation filtering logic!

**Comment by dwbrown2** [comment]
@SheepGotoHeaven I actually just looked at the frontend, and I think this may already be supported. Would you be able to take a look when you have a minute?

**Comment by SheepGotoHeaven** [comment]
Sure, I'll take a look today 👍

**Comment by SheepGotoHeaven** [comment]
To clarify, are we talking about Assets or Allocations? I don't think the Assets page currently supports a `namespace` filter. The rest of this is about Allocations filtering, which I believe works similarly to Assets.

In terms of AND / OR logic, the API will always OR together filters of the same field (e.g., if you supply filters for `namespace: "X"` and `namespace: "Y"`, you will get a response containing the union of items in namespaces X and Y), and AND together filters of different fields (e.g. `namespace: "X", service: "Y" -> intersection of namespace X and service Y`).

In the current UI there's an extra limitation, which is that we don't even OR same-field filters. It looks like we just replace the previous filter (e.g. if you apply a filter for `namespace: "X"` and then add a filter for `namespace: "Y"`, the filter for `namespace: "X"` is removed.

We should definitely at least stop dropping a filter in the UI. In terms of API behavior...I think this is another point to push us towards having the big conversation about how complex we want filtering to get. Does it actually make sense to allow ANDing on a namespace? The OP here implies that's desirable, and I'm a total k8s noob so I'm probably wrong here, but AFAIK a pod *can't* exist in 2 different namespaces and the resultant query would always be empty. There are certainly cases where it makes sense, though, like being able to query items on an intersection of labels.

Based on how often we're asked about this, it seems important to expand out capability here soon. The question is how to expand it to cover most user needs without trying to be DataDog 😅 


### TL;DR: We don't support this today and should prioritize expanding filer complexity soon.

**Comment by Adam-Stack-PM** [comment]
This issue has been marked as stale because it has not had recent activity. It will be closed if no further action occurs.
